fairroot/online/steer/FairRunOnline.cxx

```diff
@@ -217,12 +217,18 @@ Int_t FairRunOnline::EventLoop()
     auto const tmpId = GetEvtHeaderRunId();
 
     if (tmpId != fRunId) {
-        LOG(info) << "FairRunOnline::EventLoop() Call Reinit due to changed RunID (from " << fRunId << " to " << tmpId;
         fRunId = tmpId;
-        Reinit(fRunId);
-        if (!GetSource()->ReInitUnpackers()) {
-            LOG(fatal) << "FairRunOnline->EventLoop() ReInitUnpackers() failed!";
-            return 1;
+        if (!fStatic) {
+            LOG(info) << "FairRunOnline::EventLoop() Call Reinit due to changed RunID (from " << fRunId << " to "
+                      << tmpId << ")";
+            Reinit(fRunId);
+            if (!GetSource()->ReInitUnpackers()) {
+                LOG(fatal) << "FairRunOnline->EventLoop() ReInitUnpackers() failed!";
+                return 1;
+            }
+        } else {
+            LOG(info) << "FairRunOnline::EventLoop() Changed RunID (from " << fRunId << " to " << tmpId
+                      << "). ReInit not called because initialisation is static.";
         }
         fTask->ReInitTask();
```

Comment:
FairRunAna has the `fTask->ReInitTask();` inside the `if(!fStatic)`?

See for example:

https://github.com/FairRootGroup/FairRoot/blob/76fc0d676477f188843af073e74fa2ffd789db09/fairroot/base/steer/FairRunAna.cxx#L348-L351

So I wonder, why this is not consistant here?

If there is a reason, please put a comment about that reason here.

Reply:
Fixed the commit to include `ReInitTask()` in the `!fStatic` case.
Now the comments should print correct values.